src/views/Pools/components/CakeVaultCard/VaultStakeModal.tsx

```diff
@@ -152,9 +152,12 @@ const VaultStakeModal: React.FC<React.PropsWithChildren<VaultStakeModalProps>> =
     const receipt = await fetchWithCatchTxError(() => {
       // .toString() being called to fix a BigNumber error in prod
       // as suggested here https://github.com/ChainSafe/web3.js/issues/2077
-      return isWithdrawingAll
-        ? callWithGasPrice(vaultPoolContract, 'withdrawAll', undefined, callOptions)
-        : callWithGasPrice(vaultPoolContract, 'withdrawByAmount', [convertedStakeAmount.toString()], callOptions)
+      if (isWithdrawingAll) {
+        return callWithGasPrice(vaultPoolContract, 'withdrawAll', undefined, callOptions)
+      }
+
+      const methodsName = pool.vaultKey === VaultKey.CakeVault ? 'withdrawByAmount' : 'withdraw'
+      return callWithGasPrice(vaultPoolContract, methodsName, [convertedStakeAmount.toString()], callOptions)
```

Comment:
@chef-jojo @memoyil 
Do you guys remember how to calculate the number of withdrawn shares?
Flexible cake withdraw props is `shares` not `amount`